### PR TITLE
Add component scanning to external beans XML which can be overridden

### PR DESCRIPTION
**Pull request by robertgmoss**

### CODE REVIEW

File: brooklyn-tosca-transformer/src/main/resources/external-context.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns:context="http://www.springframework.org/schema/context"
+       xsi:schemaLocation="
+        http://www.springframework.org/schema/beans
+        http://www.springframework.org/schema/beans/spring-beans.xsd
+        http://www.springframework.org/schema/context
+        http://www.springframework.org/schema/context/spring-context.xsd">
+
+    <context:component-scan base-package="io.cloudsoft.tosca.a4c.brooklyn.spec">
+        <context:exclude-filter type="assignable" expression="io.cloudsoft.tosca.a4c.brooklyn.spec.RuntimeEnvironmentModifier"/>
```
Comment (sjcorbett) [inline brooklyn-tosca-transformer/src/main/resources/external-context.xml]: Include a comment indicating why this is disabled.

File: brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/platform/Alien4CloudSpringConfig.java [hunk 2]
```diff
@@ -21,6 +21,7 @@
         "org.elasticsearch.mapping",
         "io.cloudsoft.tosca.a4c.brooklyn"
 }, excludeFilters = {
+        @ComponentScan.Filter(type = FilterType.REGEX, pattern = "io.cloudsoft.tosca.a4c.brooklyn.spec.*"),
```
Comment (sjcorbett) [inline brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/platform/Alien4CloudSpringConfig.java:24]: What happens if this is not excluded?

File: brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/platform/Alien4CloudSpringContext.java [hunk 3]
```diff
@@ -11,9 +11,11 @@
 import org.springframework.context.ApplicationContext;
 import org.springframework.context.annotation.AnnotationConfigApplicationContext;
 import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.ImportResource;
 import org.springframework.core.env.PropertiesPropertySource;
 
 @ComponentScan
+@ImportResource("classpath:/external-context.xml")
```
Comment (sjcorbett) [inline brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/platform/Alien4CloudSpringContext.java]: `base-context`?

File: brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/platform/Alien4CloudSpringConfig.java (hunk 2 above)
Comment (robertgmoss) [inline brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/platform/Alien4CloudSpringConfig.java:24]: All are scanned and loaded before the external config gets a chance to exclude

**Comment by sjcorbett** [comment]
test-context.xml needs to be updated too, I presume.

**Comment by robertgmoss** [comment]
@sjcorbett thanks, comments addressed

File: brooklyn-tosca-transformer/src/test/resources/test-context.xml [hunk 4]
```diff
@@ -16,17 +16,20 @@
         <!-- alien4cloud.security.* to exclude ldap package is not working in component-scan -->
         <context:exclude-filter type="regex" expression="alien4cloud\.security.*"/>
         <context:exclude-filter type="regex"
-                expression="alien4cloud\.ldap.*|alien4cloud\.initialization.*|alien4cloud\.documentation.*|alien4cloud\.audit.*|alien4cloud\.security\.users\.rest.*|alien4cloud\.security\.groups\.rest.*"/>
+                                expression="alien4cloud\.ldap.*|alien4cloud\.initialization.*|alien4cloud\.documentation.*|alien4cloud\.audit.*|alien4cloud\.security\.users\.rest.*|alien4cloud\.security\.groups\.rest.*"/>
```
Comment (sjcorbett) [inline brooklyn-tosca-transformer/src/test/resources/test-context.xml]: Can you get your IDE to stop formatting lines you haven't changed?

File: brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/ApplicationSpecsBuilder.java [hunk 5]
```diff
@@ -94,6 +94,7 @@ public ApplicationSpecsBuilder(EntitySpecFactory entitySpecFactory, Collection<E
         visited.add(node);
 
         EntitySpec<?> spec = createSpec(node, nodeTemplate, topology);
+        LOG.info("applying spec modifiers {}", specModifiers);
```
Comment (sjcorbett) [inline brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/ApplicationSpecsBuilder.java]: This is debug at best, probably trace, and should include the spec/nodeTemplate in question. One to tweak subsequently.

**Pull request merged**